s3_scrubber/src/checks.rs

```diff
@@ -107,33 +113,62 @@ pub(crate) async fn branch_cleanup_and_check_errors(
                                         ))
                         }
 
-                        if !s3_layers.remove(&layer) {
+                        let layer_map_key = (layer, metadata.generation);
+                        if !s3_layers.remove(&layer_map_key) {
+                            // FIXME: this will emit false positives if an index was
+                            // uploaded concurrently with our scan.  To make this check
+                            // correct, we need to try sending a HEAD request for the
+                            // layer we think is missing.
                             result.errors.push(format!(
-                                "index_part.json contains a layer {} that is not present in S3",
-                                layer.file_name(),
+                                "index_part.json contains a layer {}{} that is not present in S3",
```

Comment:
Should we maybe use "remote storage" instead of S3 here because we now support azure blob storage?

Reply:
hmmm thinking of it this is probably a larger change, because it would involve a rename of the entire tool. so it's okay if we do it later.

Reply:
Updated this anyway, since we're moving in that direction.